packages/web-components/@groupby/elements-products/test/unit/products.test.ts

```diff
@@ -152,26 +222,56 @@ describe('Products Search Component', () => {
   itShouldExtendClass(() => component, ProductsBase);
 
   describe('connectedCallback', () => {
+    let cacheResponseEventName;
+
+    beforeEach(() => {
+      cacheResponseEventName = component.cacheResponseEventName = 'sayt-cache-response';
+    });
+
     itShouldCallParentMethod(() => component, 'connectedCallback');
 
-    it('should set up an event listener for a search response to set products', () => {
-      const addEventListener = sinon.stub(window, 'addEventListener');
+    it('should add event listeners for sayt products events', () => {
+      const addEventListener = stub(window, 'addEventListener');
 
       component.connectedCallback();
 
-      expect(addEventListener).to.be.calledWith(SEARCH_RESPONSE, component.setProductsFromEvent);
+      expect(addEventListener).to.be.calledWith(
+        SEARCH_RESPONSE,
+        component.setProductsFromEvent
+      );
+      expect(addEventListener).to.be.calledWith(
+        cacheResponseEventName,
+        component.setProductsFromCacheEvent
+      );
+    });
+
+    it('should request initial data', () => {
+      const requestInitialData = stub(component, 'requestInitialData');
+
+      component.connectedCallback();
+
+      expect(requestInitialData).to.be.calledWith(SEARCH_RESPONSE, component.group, cacheResponseEventName);
     });
   });
 
   describe('disconnectedCallback', () => {
     itShouldCallParentMethod(() => component, 'disconnectedCallback');
 
-    it('should remove an event listener for search responses', () => {
+    it('should remove event listeners for sayt products', () => {
       const removeEventListener = sinon.stub(window, 'removeEventListener');
```

Comment:
Since you're modifying the code around it anyway, we may as well normalize this now.
```suggestion
      const removeEventListener = stub(window, 'removeEventListener');
```